api/server/app.cc

```diff
@@ -456,34 +458,60 @@ static const char *SrcCgroup() {
     return NULL;
 }
 
+int setCgroups(std::string file1, std::string file2){        
+	std::ifstream ifs(file1);
+	std::ofstream ofs(file2);
+    if((!ifs.is_open()) || (!ofs.is_open())){
+	return -1;
+    } else {
+	ofs << ifs.rdbuf();
+	return 0;
+    }
+}
+
 static int InitCgroup(int multiplier) {
     const char *cgroupPath = SrcCgroup();
-
+    int check = 0;
+    
     if (!cgroupPath)
         return -1;
-    std::string create_dir("mkdir " + std::string(cgroupPath) + "/butterfly");
-
-    BASH(create_dir) {
-        LOG_WARNING_("can't create butterfly cgroup, fail cmd '%s'",
-                     create_dir.c_str());
+    std::string directory = std::string(cgroupPath) + "/butterfly" ; 
+    check = mkdir((char*)directory.c_str(),S_IWGRP );
+    if (check != 0){
```

Comment:
```
if (mkdir((char*)directory.c_str(),S_IWGRP ) < 0)
```
so you can remove check variable, and `< 0`, is more used than `!= 0`, it's the same in this case, but it's common in C to have all negative value treat as errors.